movedata/move.go

```diff
@@ -71,31 +94,32 @@ func (mover *Mover) MoveData() error {
 	if err != nil {
 		return err
 	}
+
 	for rows.Next() {
 		columns := make([]interface{}, len(cols))
-		columnPointers := make([]interface{}, len(cols))
 		for i := range columns {
-			columnPointers[i] = &columns[i]
+			columns[i] = &columns[i]
 		}
 
-		if err := rows.Scan(columnPointers...); err != nil {
+		if err := rows.Scan(columns...); err != nil {
 			return err
 		}
-
+		convertedArrayOfData := casting.CastArrayOfData(columnsMap, cols, columns)
 		document := make(map[string]interface{})
 		for i, colName := range cols {
-			val := columnPointers[i].(*interface{})
-			document[colName] = *val
+			document[colName] = convertedArrayOfData[i]
 		}
-
 		for _, jsonfield := range mover.queryConf.JSONFields {
 			var jsonData map[string]interface{}
+
 			byteData := document[jsonfield.FieldName].([]byte)
-			//TODO consider also the field types in parsing
 			err := json.Unmarshal(byteData, &jsonData)
 			if err != nil {
 				return err
 			}
+			for _, field := range jsonfield.Fields {
+				jsonData[field.Name] = casting.CastSingleElement(jsonColsMap[jsonfield.FieldName], field.Name, jsonData[field.Name])
```

Comment:
you should check both

```golang
jsonData[field.Name]
```
and
```golang
 document[jsonfield.FieldName]
```

with

```golang
v ,ok := map["key"]
```golang

Reply:
in case they are not there, we should return an user friendly error that will be logged outside this function

Reply:
This is true for the value extracted from map (I applied it) but the type conversion does not return an error, the fallback is always managed.